src/processpriv.cpp

```diff
@@ -206,167 +289,234 @@ boost::filesystem::path ProcessPriv::optsPath(const char* command, const QoreHas
     return boost::filesystem::absolute(ret);
 }
 
-int ProcessPriv::exitCode(ExceptionSink *xsink) {
+void ProcessPriv::processArgs(const QoreListNode* arguments, std::vector<std::string>& out) {
+    if (arguments) {
+        for (qore_size_t i = 0; i < arguments->size(); i++) {
+            QoreStringNodeValueHelper s(arguments->retrieveEntry(i));
+            out.push_back(s->getBuffer());
+        }
+    }
+    else {
+        out.push_back(""); // just a dummy argument to get it working inside child()
```

Comment:
I think it's needed for the bp::child construction on line 135. If the vector was empty it would probably not work and therefore another construction would have to be done in an if-else block. Like this, there's only one construction code. I'll at least make it look nicer.